lib/utils/hfbBuffer.hpp

```diff
@@ -0,0 +1,272 @@
+/*****************************************
+@file
+@brief Code for using the hfbBuffer formatted data.
+- hfbMetadata
+- hfbFrameView
+*****************************************/
+#ifndef HFBBUFFER_HPP
+#define HFBBUFFER_HPP
+
+#include "buffer.h"
+#include "chimeMetadata.h"
+#include "visUtil.hpp"
+
+#include "gsl-lite.hpp"
+
+#include <complex>
+#include <set>
+#include <sys/time.h>
+#include <time.h>
+#include <tuple>
+
+
+/**
+ * @brief The fields within the hfbBuffer.
+ *
+ * Use this enum to refer to the fields.
+ **/
+enum class hfbField { hfb, weight, flags, gain };
+
+
+/**
+ * @struct hfbMetadata
+ * @brief Metadata for the hyperfine beam style buffers
+ *
+ * @author James Willis
+ **/
+struct hfbMetadata {
+
+    /// The FPGA sequence number of the integration frame
+    uint64_t fpga_seq_start;
+    /// The ctime of the integration frame
+    timespec ctime;
+    /// Nominal length of the frame in FPGA ticks
+    uint64_t fpga_seq_length;
+    // Amount of data that actually went into the frame (in FPGA ticks)
+    uint64_t fpga_seq_total;
+
+    /// ID of the frequency bin
+    uint32_t freq_id;
+    /// ID of the dataset (vis, gatedvisX ...), main vis dataset = 0
+    uint64_t dataset_id;
+
+    /// Number of elements for data in buffer
+    uint32_t num_elements;
+    /// Number of products for data in buffer
+    uint32_t num_prod;
```

Comment:
I would suggest breaking this one down into `num_beams` and `num_subfreq` (or similar) parameters. I can see it being very useful having that information available even if you don't directly need it to decode the frame.